docs/scos/dev/front-end-development/202212.0/oryx/i18n/localization.md

```diff
@@ -0,0 +1,138 @@
+
+```ts
+import { appBuilder } from "@spryker-oryx/application";
+import { I18nFeature } from "@spryker-oryx/i18n";
+
+export const app = appBuilder()
+  .withFeature(
+    new I18nFeature({
+      // Here as example we are loading translations from TS modules
+      load: (localeId) => import(`../i18n/${localeId}.ts`), // <-- Required part
+    })
+  )
+  .create();
+```
+
+This gives you full flexibility to how and where to load your translation texts from. The `load` function returns a promise of a `{ default: I18nData }` type.
+
+## Translation resources
+
+In order to translate text in Oryx, you can provide translation data in a TS module. You can also integrate those in data provided by a 3rd party translation engine.
+
+```ts
+export default {
+  increase: "Increase",
+
+  // using token parameters
+  "cart.entry.<quantity>-items": "x {quantity}",
+
+  // pluralization example
+  "order.<count>-items":
+    "Products ({count, plural, one {{count} item} other {{count} items}})",
+};
+```
+
+The translation engine is based on standardized [ICU message expressions](https://unicode-org.github.io/icu/userguide/format_parse/messages/). The ICU message format defines a standard syntax to handle common translation cases:
+
+- Pluralization: for example, `You have {count} {count, plural, one {item} other {items}} in cart`.
+- Selects: for example, `{gender, select, male {He} female {She} other {They}} invited you to party!`.
+
+Other ICU capabilities, such as [number formatting](https://unicode-org.github.io/icu/userguide/format_parse/numbers/) or [date and time formatting](https://unicode-org.github.io/icu/userguide/format_parse/datetime/) can be loaded in addition. Most likely this is not needed in your project, since Oryx uses the standard [Intl](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl) api that is widely supported.
+
+## Use the i18n directive in components
+
+Localized content is using [the fine-grained reactivity system](/docs/scos/dev/front-end-development/{{page.version}}/oryx/reactivity/reactivity.html). The localizations are updated as soon the active locale is changed, for example when the user uses the localization selector or when you use the `LocaleService.set()` api in code. All localized content in Oryx components is updated instantly without a page reload.
+
+To support such fine-grained reactivity, a frameworks specific implementation is required. Oryx provides a `i18n` lit-directive that is used inside components. Using this directive, ensures that the DOM is aligned with the localizations in an efficient way. This requires the `@signalAware()` decorator on the component class. To simplify the integration, you can leverage the `I18nMixin` in your component implementation that will add the `@signalAware()` and expose the `i18n` function as a method on the component.

```

Comment:
> This requires the `@signalAware()` decorator on the component class.

There are can be some questions on this point probably, maybe we can explain it a bit?

Reply:
Agreed. I've left a TODO to link to signals once it's ready.